### ISSUE TITLE
Migrate from GreenDao to something else

### ISSUE THREAD

**Issue by Vinetos** [Vinetos/Tranquille#6]
GreenDAO is not FOSS as the gradle plugin is still not opens-sourced.
https://github.com/greenrobot/greenDAO/issues/412
We have to select another database library in order to be FOSS and be published on F-Droid

**Comment by Penaz91** [comment]
Hello! I have been following the development of this fork since I saw your post on YACB's gitlab. Seems that another repository I follow has run onto the same issue as yours: https://github.com/breezy-weather/breezy-weather/issues/677

In this precise case, SQLDelight was used, but the issue lists some alternatives that you may want to check out. Hopefully this helps a bit!

**Comment by Vinetos** [comment]
Hello !
Thank you for letting me know. I am currently looking for an alternative that could match the current behaviour. 

I will have a look at this issue to see how they resolved this !

**Comment by Vinetos** [comment]
SQLDelight seems to be an good alternative and make the transition path easy enough. I think this it what we will use.

**Comment by Vinetos** [comment]
I am pinning this issue as it takes me a lot of work before doing other one !

**Comment by licaon-kter** [comment]
Can you drop the plugin and include the generated part, eg: https://github.com/wallabag/android-app/pull/1406

**Comment by Vinetos** [comment]
Yep, that something I will do very soon.

**Comment by linsui** [comment]
The source code is published.

**Comment by Vinetos** [comment]
If the plugin is effectively the one used, migration to the open-srouced version will be easier as YACB is strongly linked to greenDAO work.  
I have notified F-Droid team that greenDAO is open-sourced, if the app is then accepted to the store, I will close this issue or mark it as "good to have one day".

**Comment by licaon-kter** [comment]
@Vinetos you do know that https://github.com/greenrobot/greenDAO (the actual reason for the plugin) is no longer developed right?

**Comment by Vinetos** [comment]
Indeed, transitioning to SQLDelight is on my agenda, albeit it's more of a desirable enhancement rather than an immediate necessity. In my perspective, the majority of the work for SQLDelight has been accomplished. However, due to time constraints, I haven't been able to finalize and thoroughly test the migration process.

Probably a two step transition is possible and easier to perform :
1 - remove greendao and use generated models as-it-is
2 - add sqldelight for a better database management

**Comment by Vinetos** [comment]
Done with #16
